### PR TITLE
front: bump ui-core and ui-speedspacechart to 46

### LINKED ISSUE #8333: Front: Speed space chart is broken when there is no stop at the end
### What happened?

When we put no stop at the end, the GEV is broken : 

Image

### What did you expect to happen?

We should have a correct GEV

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a TrainSchedule
2. Remove the stop at the end

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

staging 0a34296

### LINKED ISSUE #8885: Add MRSP for GEV
## Mock-up

2024-09-16-152121_1439x719_scrot

## AC for osrd-ui

- [x] Red limits are standard speed limits (any speed limit except the temporary ones)
- [x] Yellow ones are temporary speed limits
- [x] The train length before accelerations must be displayed as the mockup specifies
- [x] When the checkbox is ticked, both temporary and normal speed limits are display
- [ ] ~~There must be two new checkboxes in the settings to display the standard and temporary speed limits~~
- [ ] ~~Those checkboxes can be ticked independentely~~
- [ ] ~~If only Speed limits is ticked, only the standard speed limits are displayed (in red). There must be a value of speed limit everywhere along the path.~~
- [ ] ~~If only the Temporary speed restrictions is ticked, only the temporary speed limits are displayed (in yellow). Nothing is displayed if there's no temporary speed limit.~~
- [ ] ~~If both are ticked, the most restrictive speed limit si displayed with the right color corresponding to the speed limit type~~

## AC for osrd (integration)

- [ ] The MRSP is implemented in operational studies (no temporary speed limits)
- [ ] The MRSP is implemented in stdcm (no temporary speed limits yet, require backend)

### LINKED ISSUE #9160: front: bump ui-core, ui-speedspacechart and ui-icons to 0.0.46
### Description and goal

bump ui-core, ui-speedspacechart and ui-icons to 0.0.46:
- displays the correct mrsp on the GEV
- allow the user to type in the DatePicker in stdcm

### Acceptance criteria

- [ ] the mrsp in the GEV is correct
- [ ] the user can type in the datePicker input
- [ ] if the user enter a bad input in the DatePicker, an error is shown

### PR DESCRIPTION
**Pull request by clarani**
closes #9160
closes #8885
closes https://github.com/OpenRailAssociation/osrd/issues/8333

- displays the correct mrsp on the GEV
- allow the user to type in the DatePicker in stdcm

## To test
- [ ] the mrsp in the GEV is correct (activate the speedLimit option)
image
- [ ] the user can type in the datePicker input
image
- [ ] if the user enters a bad input in the DatePicker, an error is shown (we keep this display for now, it will be improved in an other PR (validated by Thibaut))
image
image

### CODE REVIEW

**Comment by emersion** [comment]
Can all of the commits be squashed together? The intermediate state doesn't build because all osrd-ui modules must always be at the exact same version.

Hm, or do they…?

(Typo in the commit message: s/46/45/)

**Comment by clarani** [comment]
> Can all of the commits be squashed together? The intermediate state doesn't build because all osrd-ui modules must always be at the exact same version.
> Hm, or do they…?

No they don't :) The intermediate states build :)

> (Typo in the commit message: s/46/45/)

The names of my previous commits were not correct, I fixed them

**Review by flomonster — CHANGES_REQUESTED** [review]
Weird behavior when the last range is smaller than the train length :shrug: 

image

> [!NOTE]
> This is actually an osrd-ui bug. We can ignore it for now. PR [here](https://github.com/OpenRailAssociation/osrd-ui/pull/601)

**Review by flomonster — APPROVED** [review]
LGTM

**Review by RomainValls — APPROVED** [review]
LGTM and Tested 👍

File: front/src/utils/date.ts [hunk 1]
```diff
@@ -241,3 +241,10 @@ export const generateISODateFromDateTime = ({ date, hours, minutes }: ScheduleCo
   date.setMinutes(minutes);
   return formatLocaleDateToIsoDate(date);
 };
+
+/** Format a date to a string 'DD/MM/YY', for instance '01/01/24' */
+export const formatDateString = (date?: Date | null) => {
+  if (!date) return '';
+  const dayjsDate = dayjs(date);
+  return dayjsDate.format('DD/MM/YY');
```
Comment (Math-R) [inline front/src/utils/date.ts]: why not just ` return dayjs(date).format('DD/MM/YY')` ?

File: front/public/locales/en/stdcm.json [hunk 2]
```diff
@@ -14,6 +14,12 @@
   },
   "departureTime": "Departure time",
   "destinationTime": "Destination time",
+  "form": {
+    "datePickerErrors": {
+      "invalidInput": "The date must be formatted as dd/mm/yy",
+      "invalidDate": "Select a date bewteen le {{startDate}} and le {{endDate}}"
```
Comment (Math-R) [inline front/public/locales/en/stdcm.json]: typo here bewteen must be between

File: front/src/utils/date.ts (hunk 1 above)
Comment (clarani) [reply front/src/utils/date.ts]: Done

**Review by Math-R — APPROVED** [review]
lgtm :)

**Pull request merged**